internal/cloudproviders/aws/awsgo.go

```diff
@@ -0,0 +1,889 @@
+
+	AuthorizeSecurityGroupEgress(ctx context.Context, ec2Client *ec2.Client, parameter ec2.AuthorizeSecurityGroupEgressInput) error
+
+	ImportKeyPair(ctx context.Context, ec2Client *ec2.Client, keypair *ec2.ImportKeyPairInput) error
+
+	ModifyVpcAttribute(ctx context.Context, ec2client *ec2.Client) error
+	ModifySubnetAttribute(ctx context.Context, ec2client *ec2.Client) error
+
+	SetRegion(string) string
+	SetVpc(string) string
+}
+
+type AwsGoClient struct {
+	ACESSKEYID     string
+	ACESSKEYSECRET string
+	Region         string
+	Vpc            string
+}
+
+func (*AwsGoClient) AuthorizeSecurityGroupIngress(ctx context.Context, ec2Client *ec2.Client, parameter ec2.AuthorizeSecurityGroupIngressInput) error {
+
+	_, err := ec2Client.AuthorizeSecurityGroupIngress(ctx, &parameter)
+	if err != nil {
+		log.Error("Error Authorizing Security Group Ingress", "error", err)
+	}
+
+	return nil
+}
+
+func (*AwsGoClient) BeginCreateNIC(ctx context.Context, ec2client *ec2.Client, parameter *ec2.CreateNetworkInterfaceInput) (*ec2.CreateNetworkInterfaceOutput, error) {
+
+	nic, err := ec2client.CreateNetworkInterface(ctx, parameter)
+	if err != nil {
+		log.Error("Error Creating Network Interface", "error", err)
+	}
+
+	nicExistsWaiter := ec2.NewNetworkInterfaceAvailableWaiter(ec2client, func(nicwaiter *ec2.NetworkInterfaceAvailableWaiterOptions) {
+		nicwaiter.MinDelay = 20
+		nicwaiter.MaxDelay = 60
+	})
+
+	describeNICInput := &ec2.DescribeNetworkInterfacesInput{
+		NetworkInterfaceIds: []string{*nic.NetworkInterface.NetworkInterfaceId},
+	}
+
+	err = nicExistsWaiter.Wait(context.Background(), describeNICInput, 60*time.Second)
+
+	if err != nil {
+		log.Error("Error Waiting for Network Interface", "error", err)
+	}
+
+	return nic, err
+}
+
+func (*AwsGoClient) BeginCreatePubIP() error {
+	panic("unimplemented")
+}
+
+func (*AwsGoClient) BeginCreateSecurityGrp() error {
+	panic("unimplemented")
+}
+
+func (awsclient *AwsGoClient) BeginCreateSubNet(context context.Context, subnetName string, ec2client *ec2.Client, parameter ec2.CreateSubnetInput) (*ec2.CreateSubnetOutput, error) {
+	subnet, err := ec2client.CreateSubnet(context, &parameter)
+	if err != nil {
+		log.Error("Error Creating Subnet", "error", err)
+	}
+
+	subnetExistsWaiter := ec2.NewSubnetAvailableWaiter(ec2client, func(subnetwaiter *ec2.SubnetAvailableWaiterOptions) {
+		subnetwaiter.MinDelay = 20
+		subnetwaiter.MaxDelay = 60
+	})
+
+	describeSubnetInput := &ec2.DescribeSubnetsInput{
+		SubnetIds: []string{*subnet.Subnet.SubnetId},
+	}
+
+	err = subnetExistsWaiter.Wait(context, describeSubnetInput, 60*time.Second)
+	if err != nil {
+		log.Error("Error Waiting for Subnet", "error", err)
+	}
+
+	return subnet, err
+}
+
+func (*AwsGoClient) BeginCreateVM(ctx context.Context, ec2client *ec2.Client, parameter *ec2.RunInstancesInput) (*ec2.RunInstancesOutput, error) {
+
+	printcounter.Do(func() {
+		time.Sleep(2 * time.Second)
+		log.Note("Creating ec2 instances......")
+	})
```

Comment:
I am not sure whether it can somehow benefit in logs

But still pretty impressive findings. would need to understand when we would need this

Reply:
we cant directly print if we then in each go routine it will print `creating vms....`  thats why used `sync.Once`

Reply:
so is there a creating e2 instance :id ?
message>?

Reply:
I think we can remove it completely as I am pretty sure aws/vm.go prints creating vm with name